### PR TITLE
wifi: Add twt traffic gen support

### PR DESCRIPTION
**Pull request by chiranjeevi2776**
Add generic traffic generation module and use that in the TWT sample to exchange data after TWT flow is established. This gives us fine grained control that will be used in upcoming enhancements where the traffic will be synchronized to TWT service periods.

### CODE REVIEW

File: samples/wifi/twt/README.rst [hunk 1]
```diff
@@ -22,6 +22,21 @@ Overview
 The sample can perform Wi-Fi operations such as connect and disconnect in the 2.4 GHz and 5 GHz bands depending on the capabilities of an access point.
 
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode and setup TWT flow with the access point.
+By default once the TWT flow is setup, the sample will use the traffic generator module to send and receive TCP data packets to and from the traffic generator
+server running on a PC connected to the same access point (typically via Ethernet).
```
Comment (kittydepa) [inline samples/wifi/twt/README.rst:26]:
```suggestion
server running on a PC connected to the same access point (typically through Ethernet).
```
Please also combine this part with the link above (so the same sentence is all on the same line).

File: samples/wifi/twt/README.rst [hunk 2]
```diff
@@ -22,6 +22,21 @@ Overview
...
+
+Traffic generator server
+************************
+When traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically via Ethernet).
```
Comment (kittydepa) [inline samples/wifi/twt/README.rst:30]:
```suggestion

When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically through Ethernet).
```

File: samples/wifi/twt/README.rst [hunk 3]
```diff
@@ -22,6 +22,21 @@ Overview
...
+The server source is available in the :file:`scripts/traffic_gen_server` file. This is a `python3` based server, so, you need to have `python3` installed on your PC.
```
Comment (kittydepa) [inline samples/wifi/twt/README.rst:31]:
```suggestion
The server source is available in the :file:`scripts/traffic_gen_server` file.
This is a `python3` based server, so, you need to have `python3` installed on your PC.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]

**Review by krish2718 — CHANGES_REQUESTED** [review]
Manifest update was removed in latest push. Please fix

**Review by krish2718 — APPROVED** [review]

File: samples/wifi/twt/README.rst [hunk 4]
```diff
@@ -22,6 +22,23 @@ Overview
 The sample can perform Wi-Fi operations such as connect and disconnect in the 2.4 GHz and 5 GHz bands depending on the capabilities of an access point.
 
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode and setup TWT flow with the access point.
+By default once the TWT flow is setup, the sample will use the traffic generator module to send and receive TCP data packets to and from the traffic generator
+server running on a PC connected to the same access point (typically through Ethernet).
```
Comment (kittydepa) [inline samples/wifi/twt/README.rst:26]: Please move this to previous line (so the entire sentence is on one line).

File: samples/wifi/twt/README.rst [hunk 5]
```diff
@@ -22,6 +22,23 @@ Overview
...
+
+Traffic generator server
+************************
+
+When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically through Ethernet).
+The server source is available in the :file:`scripts/traffic_gen_server` file.
+This is a `python3` based server, so, you need to have `python3` installed on your PC.
+
+The server can be started by running the following command:
```
Comment (richabp) [inline samples/wifi/twt/README.rst:35]:
```suggestion
To start the server, run the following command:
```

File: samples/wifi/twt/README.rst [hunk 6]
```diff
@@ -22,6 +22,23 @@ Overview
...
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode and setup TWT flow with the access point.
+By default once the TWT flow is setup, the sample will use the traffic generator module to send and receive TCP data packets to and from the traffic generator
+server running on a PC connected to the same access point (typically through Ethernet).
+
+Traffic generator server
+************************
+
+When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically through Ethernet).
+The server source is available in the :file:`scripts/traffic_gen_server` file.
+This is a `python3` based server, so, you need to have `python3` installed on your PC.
```
Comment (richabp) [inline samples/wifi/twt/README.rst:33]:
```suggestion
This is a `python3` based server, so you must make sure that `python3` is installed on your PC.
```

File: samples/wifi/twt/README.rst [hunk 7]
```diff
@@ -22,6 +22,23 @@ Overview
...
+
+The server can be started by running the following command:
+
+.. code-block:: console
+
+   python3 traffic_gen_server.py
+
+Please use the ``-h`` option to see the help message for how to use the server.
```
Comment (richabp) [inline samples/wifi/twt/README.rst:41]:
```suggestion
Use the ``-h`` option to display the help message for how to use the server.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/twt/Kconfig [hunk 8]
```diff
@@ -47,17 +50,26 @@ config TWT_SAMPLE_PASSWORD
 	help
 	  Specify the Password to connect
 
+config STA_CONN_TIMEOUT_SEC
+	int "Overall Connection timeout i.e., time to be waited for a station to connect and get an IP address"
```
Comment (richabp) [inline samples/wifi/twt/Kconfig:54]:
```suggestion
	int "Overall Connection timeout (time to be waited for a station to connect and get an IP address)"
```

File: samples/wifi/twt/README.rst [hunk 9]
```diff
@@ -22,6 +22,22 @@ Overview
...
 
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode and setup TWT flow with the access point.
+By default once the TWT flow is setup, the sample will use the traffic generator module to send and receive TCP data packets to and from the traffic generator server running on a PC connected to the same access point (typically through Ethernet).
+
+Traffic generator server
+************************
+
+When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically through Ethernet).
+The server source is available in the :file:`scripts/traffic_gen_server` file.
+This is a `python3` based server, so, you must make sure that `python3` is installed on your PC.
```
Comment (richabp) [inline samples/wifi/twt/README.rst:32]:
```suggestion
This is a `python3` based server, so you must make sure that `python3` is installed on your PC.
```

File: samples/wifi/twt/README.rst [hunk 10]
```diff
@@ -22,6 +22,22 @@ Overview
 The sample can perform Wi-Fi operations such as connect and disconnect in the 2.4 GHz and 5 GHz bands depending on the capabilities of an access point.
 
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode and setup TWT flow with the access point.
+By default once the TWT flow is setup, the sample will use the traffic generator module to send and receive TCP data packets to and from the traffic generator server running on a PC connected to the same access point (typically through Ethernet).
+
+Traffic generator server
+************************
+
+When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point (typically through Ethernet).
```
Comment (richabp) [inline samples/wifi/twt/README.rst:30]: Seems to be repetitive from line 25, so can remove it.
```suggestion
When the traffic generator module is used with this sample, the traffic generator server must be running on a PC connected to the same access point.
```

File: samples/wifi/twt/modules/traffic_gen/Kconfig [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+config TRAFFIC_GEN_SERVER
+	bool "Traffic gen module acts as a Server"
+	help
+	  Receive downlink traffic from traffic gen client (A python
+	  client running in Cloud).
+
+endchoice
+
+config TRAFFIC_GEN_DURATION
```
Comment (richabp) [inline samples/wifi/twt/modules/traffic_gen/Kconfig:28]: Can we write all Kconfig Help in a similar way to make it consistent with other Kconfigs?

**Review by richabp — APPROVED** [review]

**Review by D-Triveni — APPROVED** [review]

**Review by kittydepa — APPROVED** [review]

**Pull request merged**